server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -94,27 +109,38 @@ class LivyServer extends Logging {
 
     livyConf.get(AUTH_TYPE) match {
       case authType @ KerberosAuthenticationHandler.TYPE =>
-        val principal = SecurityUtil.getServerPrincipal(livyConf.get(KERBEROS_PRINCIPAL),
+        val principal = SecurityUtil.getServerPrincipal(livyConf.get(AUTH_KERBEROS_PRINCIPAL),
           server.host)
-        val keytab = livyConf.get(KERBEROS_KEYTAB)
+        val keytab = livyConf.get(AUTH_KERBEROS_KEYTAB)
         require(principal != null,
-          s"Kerberos auth requires ${KERBEROS_PRINCIPAL.key} to be provided.")
+          s"Kerberos auth requires ${AUTH_KERBEROS_PRINCIPAL.key} to be provided.")
         require(keytab != null,
-          s"Kerberos auth requires ${KERBEROS_KEYTAB.key} to be provided.")
+          s"Kerberos auth requires ${AUTH_KERBEROS_KEYTAB.key} to be provided.")
 
         val holder = new FilterHolder(new AuthenticationFilter())
         holder.setInitParameter(AuthenticationFilter.AUTH_TYPE, authType)
         holder.setInitParameter(KerberosAuthenticationHandler.PRINCIPAL, principal)
         holder.setInitParameter(KerberosAuthenticationHandler.KEYTAB, keytab)
         holder.setInitParameter(KerberosAuthenticationHandler.NAME_RULES,
-          livyConf.get(KERBEROS_NAME_RULES))
+          livyConf.get(AUTH_KERBEROS_NAME_RULES))
         server.context.addFilter(holder, "/*", EnumSet.allOf(classOf[DispatcherType]))
         info(s"SPNEGO auth enabled (principal = $principal)")
         if (!livyConf.getBoolean(LivyConf.IMPERSONATION_ENABLED)) {
           info(s"Enabling impersonation since auth type is $authType.")
           livyConf.set(LivyConf.IMPERSONATION_ENABLED, true)
         }
 
+        // run kinit periodically
+        val executor = new ScheduledThreadPoolExecutor(1)
+        runKinit(livyConf)
```

Comment:
This still needs to be addressed. If the first login fails, the server shouldn't come up, because more likely than not there's an error in the configuration.